tyr/servers/nginx/server.py

```diff
@@ -0,0 +1,197 @@
+from tyr.servers.server import Server
+import chef
+import requests
+import time
+import os.path
+
+class NginxServer(Server):
+
+    SERVER_TYPE = 'nginx'
+
+    CHEF_RUNLIST=['role[RoleNginx]']
+
+    def __init__(self, group=None, server_type=None, instance_type=None,
+                    environment=None, ami=None, region=None, role=None,
+                    keypair=None, availability_zone=None, security_groups=None,
+                    block_devices=None, chef_path=None):
+
+        if server_type is None: server_type = self.SERVER_TYPE
+
+        super(NginxServer, self).__init__(group, server_type, instance_type,
+                                    environment, ami, region, role,
+                                    keypair, availability_zone,
+                                    security_groups, chef_path)
+
+    def configure(self):
+
+        super(NginxServer, self).configure()
+
+        self.security_groups = [
+            'chef-nodes',
+        ]
+
+        self.IAM_ROLE_POLICIES.append('allow-describe-instances')
+        self.IAM_ROLE_POLICIES.append('allow-describe-tags')
+        self.IAM_ROLE_POLICIES.append('allow-get-nginx-config')
+        self.IAM_ROLE_POLICIES.append('allow-describe-elbs')
+
+        if self.environment == 'stage':
+            self.IAM_ROLE_POLICIES.append('allow-update-route53-stage')
+            self.IAM_ROLE_POLICIES.append('allow-modify-nginx-elbs-stage')
+            self.security_groups.append('s-nginx')
+        elif self.environment == 'prod':
+            self.IAM_ROLE_POLICIES.append('allow-update-route53-prod')
+            self.IAM_ROLE_POLICIES.append('allow-modify-nginx-elbs-prod')
+            self.security_groups.append('p-nginx')
+        self.resolve_iam_role()
+
+    def bake(self):
```

Comment:
Why redefine `bake()`?  I can see that the node's environment and run list is being set in the user data now, so it would probably just be defined twice if `bake()` wasn't redefined. First when `bake()` is called and then when the user data gets run. Which is fine, I don't think there's an issue with that. But if the `PyChef` API changed and we had to change the way that nodes were removed or added, then we would need to update both `Server` and `NginxServer`.